be/src/vec/functions/function_string.h

```diff
@@ -1576,9 +1574,10 @@ class FunctionSubstringIndex : public IFunction {
         auto& res_offsets = res->get_offsets();
         auto& res_chars = res->get_chars();
         res_offsets.resize(input_rows_count);
-
-        ColumnPtr content_column =
-                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        ColumnPtr content_column;
+        bool content_const = false;
+        std::tie(content_column, content_const) =
+                unpack_if_const(block.get_by_position(arguments[0]).column);
```

Comment:
This function requires parameters other than content to be const.
Then when the content is also const, it will go through the logic of use_default_implementation_for_constants and convert the const column into a normal column. So content column does not need to converted.